### PR TITLE
i18n(zh-cn): translate some migrate docs

### PR DESCRIPTION
**Pull request by Bluex-xx**
What kind of changes does this PR include?

- Translated content

Description
Translate the following docs into Chinese (Simplified):

- Translated [from-docusaurus.mdx](https://github.com/withastro/docs/blob/main/src/content/docs/en/guides/migrate-to-astro/from-docusaurus.mdx)
- Translated [from-eleventy.mdx](https://github.com/withastro/docs/blob/main/src/content/docs/en/guides/migrate-to-astro/from-eleventy.mdx)
- Translated [from-gatsby.mdx](https://github.com/withastro/docs/blob/main/src/content/docs/en/guides/migrate-to-astro/from-gatsby.mdx)
- Translated [from-gitbook.mdx](https://github.com/withastro/docs/blob/main/src/content/docs/en/guides/migrate-to-astro/from-gitbook.mdx)
- Translated [from-gridsome.mdx](https://github.com/withastro/docs/blob/main/src/content/docs/en/guides/migrate-to-astro/from-gridsome.mdx)

### CODE REVIEW

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+
+3. 将所有必要的 JavaScript，包括 import 语句，都放入["代码围栏" (`---`)](/zh/core-concepts/astro-components/#the-component-script)中。注意: 在 Astro 中，经常会直接在 HTML 模板内编写[条件渲染内容](/zh/core-concepts/astro-syntax/#dynamic-html)的 JavaScript。
+
+4. 使用[`Astro.props`](/zh/reference/api-reference/#astroprops)来访问之前传递给您的Gatsby函数的任何其他 props。
+
+5. 决定是否需要将导入的组件也转换为 Astro。如果已经安装了官方的 React 集成，您可以在 Astro 文件中[使用现有的React组件](/zh/core-concepts/framework-components/)。但是，如果他们不需要交互，您可能希望将它们转换为`.astro`组件！
+
+6. 删除任何 GraphQ L查询。改为使用 import 和[`Astro.glob()`](/zh/reference/api-reference/#astroglob)语句来查询您的本地文件。
+
+查看[Gatsby's Blog启动模板转换步骤示例](#引导示例-gatsby-layout-to-astro)
```
Comment (CaptainOfPhB) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx:127]: Broken link

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+
+  所有其他文件夹都是可选的，您可以按照您喜欢的方式组织您的 `src/` 文件夹的内容。Astro 项目中的其他常见文件夹包括 `src/layouts/`、`src/components`、`src/styles` 和 `src/scripts`。
+
+### 提示：将JSX文件转换为`.astro`文件
+
+以下是将 Gatsby `.js`组件转换为`.astro`组件的一些提示：
+
+1. 将现有的 Gatsby 组件函数的`return()`部分作为您的HTML模板。
+
+2. 将任何[ Gatsby 或 JSX 语法转换为 Astro 语法](#参考-转化为-astro-语法)或HTML网络标准。例如 `<Link to="">`， `{children}`和 `className`。
```
Comment (CaptainOfPhB) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx:117]: Broken link

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+
+您可能还希望从 Gatsby 的 `src/components/seo.js` 中复用代码，以便包含额外的站点元数据。注意 Astro 并不使用 `<Helmet>` 或 `<Header>`，而是直接创建 `<head>`。您可以导入和使用组件，甚至在 `<head>` 中也可以，以分离和组织您的页面内容。
+
+### 迁移页面和文章
+
+在 Gatsby 中，您的[页面和文章](/zh-cn/core-concepts/astro-pages/)可能存在于 `src/pages/` 或者在 `src` 外部的其他文件夹，如 `content`。而在 Astro 中，**所有的页面内容必须位于 `src/` 内**。
+
+#### React 页面
+
+您现有的 Gatsby JSX（`.js`）页面需要被[从 JSX 文件转换为 `.astro` 页面](#提示将jsx文件转换为astro文件)。您不能在 Astro 中使用现有的 JSX 页面文件。
```
Comment (CaptainOfPhB) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx:216]: Broken link

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+
+查看 Astro 的 [测试指南](/zh-cn/guides/testing/) 了解更多。
+
+### 重新利用配置文件
+
+Gatsby 有几个顶级配置文件，也包括站点和页面元数据，并用于路由。在您的 Astro 项目中，您不会使用任何这些 `gatsby-*.js` 文件，但可能有一些内容可以在构建 Astro 项目时重用：
+
+- `gatsby-config.js`：将您的 `siteMetadata: {}` 移动到 `src/data/siteMetadata.js`（或 `siteMetadata.json`）中，以便在页面布局中导入关于您的站点（标题，描述，社交账户等）的数据。
+
+- `gatsby-browser.js`：考虑将这里使用的任何东西直接添加到您的[主布局](#迁移布局文件)的 `<head>` 标签中。
```
Comment (CaptainOfPhB) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx:238]: Broken link

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx [hunk 5]
```diff
@@ -0,0 +1,582 @@
...
+          {` `}
+          <a href="https://www.gatsbyjs.com">Gatsby</a>
+        </footer>
+      </div>
+    )
+  }
+  export default Layout
+  ```
+
+2. 创建 `Layout.astro` 并添加此 `return` 值，[转换为 Astro 语法](#参考转换为-astro-语法)。
```
Comment (CaptainOfPhB) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx:440]: Broken link

**Review by CaptainOfPhB — COMMENTED** [review]
Please fix broken links to pass the ci check

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gridsome.mdx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+
+[Gridsome](https://gridsome.org) 是一个基于 Vue 和 GraphQL 构建的开源静态站点生成器。
+
+## Gridsome 和 Astro 之间的关键相似点
+
+Gridsome 和 Astro 有一些相似之处，这将帮助你迁移你的项目：
+
+- Gridsome 和 Astro 都是现代 Javascript 静态网站生成器，具有相似的[项目文件结构](/zh-cn/core-concepts/project-structure#directories-and-files)。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gridsome.mdx:18]:
```suggestion
- Gridsome 和 Astro 都是现代 Javascript 静态网站生成器，具有相似的[项目文件结构](/zh-cn/core-concepts/project-structure#目录和文件)。 
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-docusaurus.mdx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+- [Astro的类JSX语法](/zh-cn/core-concepts/astro-components/#组件模板)如果你习惯写 React 应该会感觉很熟悉。
+
+## Docusaurus 和 Astro 之间的主要区别
+
+当你在 Astro 中重建你的 Docusaurus 站点时，你会注意到一些重要的区别：
+
+- [MPA vs SPA](/zh-cn/concepts/mpa-vs-spa/)：Docusaurus 是一个基于React的单页应用程序(SPA)。Astro 站点是使用[`.astro`组件](/zh-cn/core-concepts/astro-components/)构建的多页应用程序，但也可以支持[React, Preact, Vue.js, Svelte, SolidJS, AlpineJS, Lit](/zh-cn/core-concepts/framework-components/)和原始 HTML 模板。
+
+- Docusaurus 设计用于构建文档网站，并且有一些内置的，特定于文档的网站功能，你需要在 Astro 中自己构建。相反，Astro 通过一个[官方的文档主题](https://github.com/withastro/astro/tree/latest/examples/docs)提供其中一些功能。这个网站就是使用该模板构建的！你还可以在我们的主题展示中找到更多具有内置功能的[社区文档主题](https://astro.build/themes?search=&categories%5B%5D=docs)。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-docusaurus.mdx:36]:
```suggestion
- Docusaurus 设计用于构建文档网站，并且有一些内置的，特定于文档的网站功能，你需要在 Astro 中自己构建。相反，Astro 通过一个[官方的文档主题](https://starlight.astro.build)提供其中一些功能。这个网站就是该模板的灵感来源！你还可以在我们的主题展示中找到更多具有内置功能的[社区文档主题](https://astro.build/themes?search=&categories%5B%5D=docs)。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-docusaurus.mdx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+- Docusaurus 站点使用MDX页面进行内容。Astro 的文档主题默认使用 Markdown(`.md`) 文件，并不需要你使用MDX。你可以选择[安装 Astro 的 MDX 集成](/zh-cn/guides/integrations-guide/mdx/)并使用`.mdx`文件作为标准Markdown文件。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-docusaurus.mdx:38]:
```suggestion
- Docusaurus 站点使用 MDX 页面进行内容。Astro 的文档主题默认使用 Markdown(`.md`) 文件，并不需要你使用 MDX。你可以选择[安装 Astro 的 MDX 集成](/zh-cn/guides/integrations-guide/mdx/)并使用`.mdx`文件作为标准 Markdown 文件。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-docusaurus.mdx [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+## 从Docusaurus切换到Astro
+
+要把一个 Docusaurus 文档站点转换为 Astro，首先使用我们的官方[Starlight 文档主题启动模板](https://starlight.astro.build)，或者在我们的[主题展示](https://astro.build/themes?search=&categories%5B%5D=docs)中探索更多社区文档主题。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-docusaurus.mdx:42]:
```suggestion
要把一个 Docusaurus 文档站点转换为 Astro，首先使用我们的官方 [Starlight 文档主题启动模板](https://starlight.astro.build)，或者在我们的[主题展示](https://astro.build/themes?search=&categories%5B%5D=docs)中探索更多社区文档主题。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-docusaurus.mdx [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+你可以向`create astro`命令传递一个`--template`参数，以一个我们的官方启动器开始一个新的Astro项目。或者，你可以[从 GitHub 上任何现有的 Astro 存储库开始一个新项目](/zh-cn/install/auto/#入门模板)。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-docusaurus.mdx:44]:
```suggestion
你可以向`create astro`命令传递一个`--template`参数，以一个我们的官方启动器开始一个新的 Astro 项目。或者，你可以[从 GitHub 上任何现有的 Astro 存储库开始一个新项目](/zh-cn/install/auto/#入门模板)。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-docusaurus.mdx [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+    ```
+    </Fragment>
+    <Fragment slot="yarn">
+    ```shell
+    yarn create astro --template starlight
+    ```
+    </Fragment>
+  </PackageManagerTabs>
+
+添加我们的 MDX 集成并带来你现有的内容文件，创建[MDX页面](/zh-cn/guides/markdown-content/)。你仍然可以通过将这些文件直接复制到 Astro 中的`src/pages/`来利用[基于文件的路由](/zh-cn/core-concepts/routing/)，这是你现在使用的同一个文件夹。创建与你现有的 Docusaurus 项目对应的名称的文件夹，你应该能保留你现有的URL。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-docusaurus.mdx:64]:
```suggestion
添加我们的 MDX 集成并带来你现有的内容文件，创建 [MDX页面](/zh-cn/guides/markdown-content/)。你仍然可以通过将这些文件直接复制到 Astro 中的`src/pages/`来利用[基于文件的路由](/zh-cn/core-concepts/routing/)，这是你现在使用的同一个文件夹。创建与你现有的 Docusaurus 项目对应的名称的文件夹，你应该能保留你现有的 URL。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-docusaurus.mdx [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+Docusaurus 可能处理了你的许多网站布局和元数据。你可能希望阅读有关[构建 Astro 布局作为Markdown和MDX的页面包装器](/zh-cn/core-concepts/layouts/#markdownmdx-布局)，看看如何在Astro中自己管理模板，包括你的页面`<head>`。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-docusaurus.mdx:66]:
```suggestion
Docusaurus 可能处理了你的许多网站布局和元数据。你可能希望阅读有关[构建 Astro 布局作为 Markdown 和 MDX 的页面包装器](/zh-cn/core-concepts/layouts/#markdownmdx-布局)，看看如何在 Astro 中自己管理模板，包括你的页面`<head>`。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+
+[Eleventy](https://11ty.dev) 是一个支持多种模板语言的开源静态站点生成器。
+
+## Eleventy (11ty) 和 Astro 的主要相似之处
+
+Eleventy (11ty) 和 Astro 有一些相似之处，这将帮助你迁移你的项目：
+
+- Astro 和 Eleventy 都是现代的、基于JavaScript的（Jamstack）网站构建工具。
+
+- Astro 和 Eleventy 都允许你使用[无头（headless）CMS，API或Markdown文件获取数据](/zh-cn/guides/data-fetching/)。你可以继续使用你喜欢的[内容创作](/zh-cn/guides/content/)系统，并能保留你现有的内容。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx:19]:
```suggestion
- Astro 和 Eleventy 都允许你使用[无头（headless）CMS，API 或 Markdown 文件获取数据](/zh-cn/guides/data-fetching/)。你可以继续使用你喜欢的[内容创作](/zh-cn/guides/content/)系统，并能保留你现有的内容。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+
+- Astro 和 Eleventy 都能构建[多页面应用](/zh-cn/concepts/mpa-vs-spa/)。
+
+## Eleventy (11ty) 和 Astro 主要差异
+
+当你在 Astro 中重建你的 Eleventy (11ty) 站点时，你会注意到一些重要的区别：
+
+- Eleventy 支持多种模板语言。Astro 支持[包含来自几个流行的JS框架（例如React，Svelte，Vue，Solid）的组件](/zh-cn/core-concepts/framework-components/)，但大部分页面模板都使用[Astro布局，页面和组件](/zh-cn/core-concepts/astro-components/)。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx:27]:
```suggestion
- Eleventy 支持多种模板语言。Astro 支持[包含来自几个流行的JS框架（例如 React，Svelte，Vue，Solid）的组件](/zh-cn/core-concepts/framework-components/)，但大部分页面模板都使用 [Astro布局，页面和组件](/zh-cn/core-concepts/astro-components/)。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+
+- Astro 在所有文件中使用[`src/`目录](/zh-cn/core-concepts/project-structure/#src)，包括站点元数据，这些文件在站点构建过程中可用于查询和处理。其中包含一个[特殊的 `src/pages/` 文件夹用于文件路由](/zh-cn/core-concepts/astro-pages/)。
+
+- Astro 使用[`public/` 文件夹存储不需要在构建过程中处理或转换的静态资源](/zh-cn/core-concepts/project-structure/#public)。
+
+- 在Eleventy中，必须手动配置CSS，JavaScript和其他资产的捆绑。[Astro为你提供开箱即用的处理方式](/zh-cn/concepts/why-astro/#易于使用)。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx:33]:
```suggestion
- 在Eleventy中，必须手动配置 CSS，JavaScript 和其他资产的打包。[Astro 为你提供开箱即用的处理方式](/zh-cn/concepts/why-astro/#易于使用)。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+
+## 从 Eleventy 切换到 Astro
+
+要将 Eleventy 博客转换为 Astro，首先从我们的博客主题启动模板开始，或者在我们的[主题展示](https://astro.build/themes/)中探索更多社区博客主题。
+
+你可以向`create astro`命令传递一个`--template`参数，以使用我们的官方启动器开始一个新的Astro项目。或者，你可以[从GitHub上的任何现有Astro库开始一个新项目](/zh-cn/install/auto/#入门模板)
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx:39]:
```suggestion
你可以向`create astro`命令传递一个`--template`参数，以使用我们的官方启动器开始一个新的 Astro 项目。或者，你可以[从 GitHub 上的任何现有 Astro 库开始一个新项目](/zh-cn/install/auto/#入门模板)
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+    ```
+    </Fragment>
+    <Fragment slot="yarn">
+    ```shell
+    yarn create astro --template blog
+    ```
+    </Fragment>
+  </PackageManagerTabs>
+
+将你现有的 Markdown（或MDX，通过我们可选的集成）文件作为内容，[创建Markdown或MDX页面](/zh-cn/guides/markdown-content/)。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx:59]:
```suggestion
将你现有的 Markdown（或MDX，通过我们可选的集成）文件作为内容，[创建 Markdown 或 MDX 页面](/zh-cn/guides/markdown-content/)。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+
+你的Eleventy项目允许你使用各种模板语言来构建你的网站。在Astro项目中，你的页面模板大部分将使用Astro组件，这些组件可以作为UI元素，布局甚至完整的页面。你可能想要探索[Astro的组件语法](/zh-cn/core-concepts/astro-components/)，看看如何使用组件在Astro中进行模板化。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx:61]:
```suggestion
你的 Eleventy 项目允许你使用各种模板语言来构建你的网站。在 Astro 项目中，你的页面模板大部分将使用Astro 组件，这些组件可以作为 UI 元素，布局甚至完整的页面。你可能想要探索[Astro 的组件语法](/zh-cn/core-concepts/astro-components/)，看看如何使用组件在 Astro 中进行模板化。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+
+要转换其他类型的网站，例如作品集或文档站点，请在[astro.new](https://astro.new)上查看更多官方启动器模板。你会找到每个项目的GitHub仓库的链接，以及一键链接到 StackBlitz，CodeSandbox 和 Gitpod 在线开发环境中的有效项目。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-eleventy.mdx:63]:
```suggestion
要转换其他类型的网站，例如作品集或文档站点，请在 [astro.new](https://astro.new) 上查看更多官方启动器模板。你会找到每个项目的 GitHub 仓库的链接，以及一键链接到 StackBlitz，CodeSandbox 和 Gitpod 在线开发环境中的有效项目。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx [hunk 20]
```diff
@@ -0,0 +1,582 @@
...
+
+## Gatsby 和 Astro 之间的关键差异
+
+当你在 Astro 中重建你的 Gatsby 站点时，你会注意到一些重要的差异：
+
+- [MPA vs SPA](/zh-cn/concepts/mpa-vs-spa/)：Gatsby 项目是 React 单页应用，并使用 `index.js` 作为你的项目的根。Astro 项目是多页站点，`index.astro` 是你的主页。
+
+- [Astro 组件](/zh-cn/core-concepts/astro-components/)不是以返回页面模板的出口函数形式编写的。相反，你将你的代码分割为专门为你的 JavaScript 设计的 "代码围栏" 和专门为你生成的 HTML 设计的主体。
+
+- [本地文件数据](/zh-cn/guides/imports/)：Gatsby 使用 GraphQL 从你的项目文件中检索数据。Astro 使用 ESM imports 和顶层等待函数（例如 [`Astro.glob()`](/zh-cn/guides/imports/#astroglob), [`getCollection()`](/zh-cn/guides/content-collections/#querying-collections)）从你的项目文件中导入数据。你可以手动向你的 Astro 项目添加 GraphQL，但默认情况下不包含它。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx:36]:
```suggestion
- [本地文件数据](/zh-cn/guides/imports/)：Gatsby 使用 GraphQL 从你的项目文件中检索数据。Astro 使用 ESM imports 和顶层 await 函数（例如 [`Astro.glob()`](/zh-cn/guides/imports/#astroglob), [`getCollection()`](/zh-cn/guides/content-collections/#querying-collections)）从你的项目文件中导入数据。你可以手动向你的 Astro 项目添加 GraphQL，但默认情况下不包含它。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx [hunk 21]
```diff
@@ -0,0 +1,582 @@
...
+
+2. 将任何[ Gatsby 或 JSX 语法转换为 Astro 语法](#参考转换为-astro-语法)或HTML网络标准。例如 `<Link to="">`， `{children}`和 `className`。
+
+3. 将所有必要的 JavaScript，包括 import 语句，都放入["代码围栏" (`---`)](/zh-cn/core-concepts/astro-components/#组件脚本)中。注意: 在 Astro 中，经常会直接在 HTML 模板内编写[条件渲染内容](/zh-cn/core-concepts/astro-syntax/#dynamic-html)的 JavaScript。
+
+4. 使用[`Astro.props`](/zh-cn/reference/api-reference/#astroprops)来访问之前传递给你的Gatsby函数的任何其他 props。
+
+5. 决定是否需要将导入的组件也转换为 Astro。如果已经安装了官方的 React 集成，你可以在 Astro 文件中[使用现有的React组件](/zh-cn/core-concepts/framework-components/)。但是，如果他们不需要交互，你可能希望将它们转换为`.astro`组件！
+
+6. 删除任何 GraphQ L查询。改为使用 import 和[`Astro.glob()`](/zh-cn/reference/api-reference/#astroglob)语句来查询你的本地文件。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx:125]:
```suggestion
6. 删除任何 GraphQL 查询。改为使用 import 和[`Astro.glob()`](/zh-cn/reference/api-reference/#astroglob)语句来查询你的本地文件。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx [hunk 22]
```diff
@@ -0,0 +1,582 @@
...
+
+以下是将 Gatsby `.js`组件转换为`.astro`组件的一些提示：
+
+1. 将现有的 Gatsby 组件函数的`return()`部分作为你的HTML模板。
+
+2. 将任何[ Gatsby 或 JSX 语法转换为 Astro 语法](#参考转换为-astro-语法)或HTML网络标准。例如 `<Link to="">`， `{children}`和 `className`。
+
+3. 将所有必要的 JavaScript，包括 import 语句，都放入["代码围栏" (`---`)](/zh-cn/core-concepts/astro-components/#组件脚本)中。注意: 在 Astro 中，经常会直接在 HTML 模板内编写[条件渲染内容](/zh-cn/core-concepts/astro-syntax/#dynamic-html)的 JavaScript。
+
+4. 使用[`Astro.props`](/zh-cn/reference/api-reference/#astroprops)来访问之前传递给你的Gatsby函数的任何其他 props。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx:121]:
```suggestion
4. 使用[`Astro.props`](/zh-cn/reference/api-reference/#astroprops)来访问之前传递给你的 Gatsby 函数的任何其他 props。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx [hunk 23]
```diff
@@ -0,0 +1,582 @@
...
+
+  所有其他文件夹都是可选的，你可以按照你喜欢的方式组织你的 `src/` 文件夹的内容。Astro 项目中的其他常见文件夹包括 `src/layouts/`、`src/components`、`src/styles` 和 `src/scripts`。
+
+### 提示：将JSX文件转换为`.astro`文件
+
+以下是将 Gatsby `.js`组件转换为`.astro`组件的一些提示：
+
+1. 将现有的 Gatsby 组件函数的`return()`部分作为你的HTML模板。
+
+2. 将任何[ Gatsby 或 JSX 语法转换为 Astro 语法](#参考转换为-astro-语法)或HTML网络标准。例如 `<Link to="">`， `{children}`和 `className`。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx:117]:
```suggestion
2. 将任何[ Gatsby 或 JSX 语法转换为 Astro 语法](#参考转换为-astro-语法)或 HTML web 标准。例如 `<Link to="">`， `{children}`和 `className`。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx [hunk 24]
```diff
@@ -0,0 +1,582 @@
...
+
+3. 将所有必要的 JavaScript，包括 import 语句，都放入["代码围栏" (`---`)](/zh-cn/core-concepts/astro-components/#组件脚本)中。注意: 在 Astro 中，经常会直接在 HTML 模板内编写[条件渲染内容](/zh-cn/core-concepts/astro-syntax/#dynamic-html)的 JavaScript。
+
+4. 使用[`Astro.props`](/zh-cn/reference/api-reference/#astroprops)来访问之前传递给你的Gatsby函数的任何其他 props。
+
+5. 决定是否需要将导入的组件也转换为 Astro。如果已经安装了官方的 React 集成，你可以在 Astro 文件中[使用现有的React组件](/zh-cn/core-concepts/framework-components/)。但是，如果他们不需要交互，你可能希望将它们转换为`.astro`组件！
+
+6. 删除任何 GraphQ L查询。改为使用 import 和[`Astro.glob()`](/zh-cn/reference/api-reference/#astroglob)语句来查询你的本地文件。
+
+查看[Gatsby's Blog启动模板转换步骤示例](#引导示例gatsby-布局转换为-astro)
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx:127]:
```suggestion
查看 [Gatsby's Blog 启动模板转换步骤示例](#引导示例gatsby-布局转换为-astro)
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx [hunk 25]
```diff
@@ -0,0 +1,582 @@
...
+
+#### React 页面
+
+你现有的 Gatsby JSX（`.js`）页面需要被[从 JSX 文件转换为 `.astro` 页面](/zh-cn/guides/migrate-to-astro/from-gatsby/#提示将jsx文件转换为astro文件)。你不能在 Astro 中使用现有的 JSX 页面文件。
+
+这些[`.astro` 页面](/zh-cn/core-concepts/astro-pages/) 必须位于 `src/pages/` 内，并且会基于它们的文件路径自动生成页面路由。
+
+#### Markdown 和 MDX 页面
+
+Astro 对 Markdown 有内建支持，并对 MDX 文件有可选集成。你现有的 [Markdown 和 MDX 文件](/zh-cn/guides/markdown-content/) 可以重用，但可能需要对它们的 frontmatter 进行一些调整，如添加 [Astro 的特殊 `layout` frontmatter 属性](/zh-cn/core-concepts/layouts/#markdownmdx-布局)。它们也可以被放置在 `src/pages/` 中以利用自动的基于文件的路由。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx:222]:
```suggestion
Astro 对 Markdown 有内置支持，并对 MDX 文件有可选集成。你现有的 [Markdown 和 MDX 文件](/zh-cn/guides/markdown-content/) 可以重用，但可能需要对它们的 frontmatter 进行一些调整，如添加 [Astro 的特殊 `layout` frontmatter 属性](/zh-cn/core-concepts/layouts/#markdownmdx-布局)。它们也可以被放置在 `src/pages/` 中以利用自动的基于文件的路由。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx [hunk 26]
```diff
@@ -0,0 +1,582 @@
...
+          <a href="https://www.astro.build">Astro</a>
+        </footer>
+      </div>
+    </body>
+  </html>
+  ```
+
+4. 添加所需的导入、props 和 JavaScript
+
+   条件性地根据页面路由和标题在 Astro 中渲染一个头部：
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx:493]:
```suggestion
   根据 Astro 中页面路由和标题条件性地渲染 header：
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx [hunk 27]
```diff
@@ -0,0 +1,582 @@
...
+  ---
+  <Layout title="主页" pathname={pagePathname}>
+      <p>Astro</p>
+  </Layout>
+  ```
+  :::tip
+  你可以[使用 `Astro.url` 获取当前页面的路径](/zh-cn/reference/api-reference/#astrourl)。
+  :::
+
+6. 为了测试条件头部，使用相同的模式创建第二个页面 `about.astro`：
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gatsby.mdx:556]:
```suggestion
6. 为了测试条件 header，使用相同的模式创建第二个页面 `about.astro`：
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gitbook.mdx [hunk 28]
```diff
@@ -0,0 +1,74 @@
+---
+title: 从 GitBook 迁移到
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gitbook.mdx:2]:
```suggestion
title: 从 GitBook 迁移到 Astro
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-gridsome.mdx [hunk 29]
```diff
@@ -0,0 +1,70 @@
...
+    ```
+    </Fragment>
+    <Fragment slot="yarn">
+    ```shell
+    yarn create astro --template blog
+    ```
+    </Fragment>
+  </PackageManagerTabs>
+
+将你现有的 Markdown 文件（或MDX文件，如果你选择了我们的可选集成）作为内容[创建Markdown或MDX页面](/zh-cn/guides/markdown-content/)。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-gridsome.mdx:60]:
```suggestion
将你现有的 Markdown 文件（或 MDX 文件，如果你选择了我们的可选集成）作为内容[创建 Markdown 或 MDX 页面](/zh-cn/guides/markdown-content/)。
```

**Review by liruifengv — CHANGES_REQUESTED** [review]

**Review by liruifengv — APPROVED** [review]
LGTM

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Amazing work 🚀

**Pull request merged**